### ISSUE TITLE
Ability to delete partially uploaded files to B2

### ISSUE THREAD

**Issue by cowwoc** [restic/restic#4096]
Output of `restic version`
--------------------------

`restic 0.14.0 compiled with go1.19 on windows/amd64`

What should restic do differently? Which functionality do you think we should add?
----------------------------------------------------------------------------------

`restic prune` (or a new command) should delete partially-uploaded files to B2

What are you trying to do? What problem would this solve?
---------------------------------------------------------

Delete unused files in order to save money

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

restic makes me happy every day :)

**Comment by cowwoc** [comment]
See https://www.backblaze.com/b2/docs/b2_list_unfinished_large_files.html. I don't think the word "large" is relevant as this returns files that are 0 bytes as well.

What's interesting is that I'm seeing some of the files in the data directory. Example: `data/cc/cc1ded63bb858f780bee96b90a862746130fcaae15c43a7579f88a95480894da`

This seems to imply that there are either unused or corrupt files in the data directory. `restic check` says nothing is wrong. It would be good to resolve this.

**Comment by MichaelEischer** [comment]
> This seems to imply that there are either unused or corrupt files in the data directory.

Nope, it just says that some upload didn't finish. That can happen when restic is interrupted or has to retry some operation. In both cases, if `check` says that the repository is fine, then it is ok.

To remove the incomplete uploads you can use e.g. the B2 command line tool https://www.backblaze.com/b2/docs/quick_command_line.html#usage and run `b2 cancel-all-unfinished-large-files <bucketName>`. I would have expected that these unfinished large files are automatically removed after some time. But I haven't found a corresponding note in the B2 documentation. 

Apparently, a similar issue also applies to S3. However, there one can at least create a lifecycle rule to delete unfinished uploads after a few days.
